dotCMS/src/main/java/com/dotcms/security/apps/AppsUtil.java

```diff
@@ -0,0 +1,150 @@
+package com.dotcms.security.apps;
+
+import com.dotcms.rest.api.v1.DotObjectMapperProvider;
+import com.dotmarketing.exception.DotDataException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.io.ByteStreams;
+import com.liferay.util.Base64;
+import com.liferay.util.Encryptor;
+import com.liferay.util.EncryptorException;
+import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.security.Key;
+import java.util.ArrayList;
+import java.util.List;
+import javax.crypto.Cipher;
+import org.apache.commons.io.input.CharSequenceInputStream;
+import org.apache.commons.lang3.ArrayUtils;
+
+public class AppsUtil {
+
+    private static final ObjectMapper mapper = DotObjectMapperProvider.getInstance().getDefaultObjectMapper();
+
+    /**
+     * Given the AppSecrets this will return a char array representing the deserialized json object.
+     * No strings are created in the transformation process.
+     * @param object
+     * @return
+     * @throws DotDataException
+     */
+    static char[] toJsonAsChars(final AppSecrets object) throws DotDataException {
+        try {
+            final byte [] bytes = mapper.writeValueAsBytes(object);
+            return bytesToCharArrayUTF(bytes);
+        } catch (IOException e) {
+            throw new DotDataException(e);
+        }
+    }
+
+    /**
+     * Takes a char array representation of a json secret and generates the domain object.
+     * No strings are created in the transformation process.
+     * @param chars
+     * @return The Secrets domain model.
+     * @throws DotDataException
+     */
+    static AppSecrets readJson(final char[] chars) throws DotDataException {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Consider using varargs for methods or constructors which take an array the last parameter.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5329810)